Shared/Samples/Add raster from service/AddRasterFromServiceView.swift

```diff
@@ -0,0 +1,83 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct AddRasterFromServiceView: View {
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    /// A Boolean value indicating whether the map is drawing.
+    @State private var mapIsDrawing = false
+    
+    /// A map with a dark gray basemap and a raster layer.
+    @State private var map: Map = {
+        let map = Map(basemapStyle: .arcGISDarkGrayBase)
+        // Creates an initial viewpoint with a coordinate point centered on
+        // San Francisco's Golden Gate Bridge.
+        map.initialViewpoint = Viewpoint(
+            center: Point(x: -13637000, y: 4550000, spatialReference: .webMercator),
+            scale: 100_000
+        )
+        // Creates a raster from an image service.
+        let imageServiceRaster = ImageServiceRaster(url: .imageServiceURL)
+        // Creates a raster layer from the raster.
+        let rasterLayer = RasterLayer(raster: imageServiceRaster)
+        map.addOperationalLayer(rasterLayer)
+        return map
+    }()
+    
+    /// The raster layer in the map.
+    private var rasterLayer: RasterLayer {
+        map.operationalLayers.first as! RasterLayer
+    }
+    
+    var body: some View {
+        MapView(map: map)
+            .onDrawStatusChanged { drawStatus in
+                // Updates the Boolean state when the map's draw status changes.
+                withAnimation {
+                    mapIsDrawing = drawStatus == .inProgress
+                }
+            }
+            .overlay(alignment: .center) {
+                if mapIsDrawing {
+                    ProgressView("Drawing…")
+                        .padding()
+                        .background(.ultraThinMaterial)
+                        .cornerRadius(10)
+                        .shadow(radius: 50)
+                }
+            }
+            .task {
+                do {
+                    // Loads a raster layer from online service.
+                    try await rasterLayer.load()
+                } catch {
+                    // Presents an error message if the raster fails to load.
+                    self.error = error
+                }
+            }
+            .errorAlert(presentingError: $error)
+    }
+}
+
+private extension URL {
+    static let imageServiceURL = URL(string: "https://gis.ngdc.noaa.gov/arcgis/rest/services/bag_bathymetry/ImageServer")!
+}
```

Comment:
You could just create this directly on line `35` rather then adding it in an extension here. Usually properties are created in extensions when they are being used in multiple spots but here it is only being used in one area. @yo1995 Is this just a way the samples prefers it?

Reply:
No preference

Reply:
I don't really either. I'll leave it up to @chriswebb09.

Reply:
What's the verdict?

Reply:
You can decide whether to make the change or not.